sharedse/src/commonJvmMain/kotlin/com/ustadmobile/sharedse/network/containeruploader/ContainerUploader.kt

```diff
@@ -0,0 +1,169 @@
+                            inStream = response.dataSrc ?: throw Exception()
+                            rangeStream = RangeInputStream(inStream, bytesSoFar.get(), bytesSoFar.get() + chunkSize)
+
+                            val remaining = fileSize - bytesSoFar.get()
+                            val sizeToRead = min(chunkSize, remaining.toInt())
+                            val buffer = ByteArray(sizeToRead)
+
+                            while (readRange < sizeToRead) {
+                                readRange += inStream.read(buffer)
+                            }
+
+                            val end = bytesSoFar.get() + readRange - 1
+
+
+                            urlConnection = URL(UMFileUtil.joinPaths(request.uploadToUrl, "/receiveData/")).openConnection() as HttpURLConnection
+                            urlConnection.connectTimeout = 5000
+                            urlConnection.doOutput = true
+                            urlConnection.requestMethod = "PUT"
+                            urlConnection.setRequestProperty("Content-Length", readRange.toString())
+                            urlConnection.setRequestProperty("Range", "bytes=${bytesSoFar.get()}-$end")
+                            urlConnection.setRequestProperty("SessionId", uploadJob.sessionId)
+                            urlConnection.outputStream.write(buffer)
+                            urlConnection.outputStream.flush()
+                            urlConnection.outputStream.close()
+                            urlConnection.connect()
+
+                            val responseCode = urlConnection.responseCode
+
+                            if (urlConnection.errorStream != null) {
+                                error = true
+                                errorMessage = String(urlConnection.errorStream.readBytes())
+                            }
+
+                            urlConnection.disconnect()
+
+
+                        } while (responseCode != HttpStatusCode.NoContent.value)
+
+                    val endedAt = bytesSoFar.get() + readRange
+                    bytesSoFar.set(endedAt)
+
+                    uploadJob.bytesSoFar = endedAt
+                    db.containerUploadJobDao.update(uploadJob)
+
+                }
+
+                downloadStatus = if (isActive) {
+                    JobStatus.COMPLETE
+                } else {
+                    JobStatus.PAUSED
+                }
+
+                uploadJob.jobStatus = downloadStatus
+                db.containerUploadJobDao.update(uploadJob)
+
+            } finally {
+                progressUpdaterJob.cancel()
+                listener?.onProgress(request, bytesSoFar.get(), contentLength.get())
+                inStream?.close()
+                rangeStream?.close()
+                urlConnection?.disconnect()
+
+            }
+            downloadStatus
+        }
+
+    }
+
+    companion object {
+
+        const val CHUNK_SIZE = 1024 * 8
```

Comment:
I would call this DEFAULT_CHUNK_SIZE (as that is what it is), and I would make it 200KB. 8KB is quite small.